pkg/graveler/ref/commit_iterator.go

```diff
@@ -110,7 +119,11 @@ func (ci *CommitIterator) Next() bool {
 	// as long as we have something in the queue we will
 	// set it as the current value and push the current commits parents to the queue
 	ci.value = heap.Pop(&ci.queue).(*graveler.CommitRecord)
-	for _, p := range ci.value.Parents {
+	parents := ci.value.Parents
+	if ci.firstParent && len(parents) >= 1 {
```

Comment:
No need to slice in case there is one. Although it is probably a bug if we find such commit.
```suggestion
	if ci.firstParent && len(parents) > 1 {
```